warehouse/query-core/src/main/java/datawave/query/predicate/PeekingPredicate.java

```diff
@@ -0,0 +1,15 @@
+package datawave.query.predicate;
+
+import com.google.common.base.Predicate;
+
+import javax.annotation.Nullable;
+
+public interface PeekingPredicate<T> extends Predicate<T> {
+    /**
+     * peek at the result of applying an input without actually applying it
```

Comment:
Lets make this clear that this is equivalent to apply unless the underlying implementation saves state affecting subsequent apply calls.  Basically peek is intended to execute apply without incurring any side effects.

Reply:
Perhaps even a pointer (using @see) denoting an instance where this actually makes a difference.